rclpy/rclpy/node.py

```diff
@@ -1649,8 +1649,10 @@ def destroy_rate(self, rate: Rate):
 
         :return: ``True`` if successful, ``False`` otherwise.
         """
-        self.destroy_timer(rate._timer)
-        rate.destroy()
+        success = self.destroy_timer(rate._timer)
+        if success:
+            rate.destroy()
+        return success
```

Comment:
If we are going to do this, we also need to change the type annotation on line 1646 to have `-> bool`.

Reply:
Fixed.